### ISSUE TITLE
Failed to perform symbolic shape inference on GPT2 Model Variants

### ISSUE THREAD

**Issue by tzhang-666** [microsoft/onnxruntime#8314]
**Describe the bug**
ONNX RUNTIME cannot perform symbolic shape inference on a valid model.

**Urgency**

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Window 10 Enterprise
- ONNX Runtime installed from (source or binary): Binary
- ONNX Runtime version: 1.8.0
- Python version: 3.8.10
- Visual Studio version (if applicable): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**To Reproduce**
- Describe steps/code to reproduce the behavior.

```Python
import onnxruntime
import torch
import onnx
import onnxruntime.tools.symbolic_shape_infer as sym
import numpy as np

if __name__ == '__main__':
    location = 'sym_shape_infer\\gpt2_full_noPast_wtoken.onnx'
    model = onnx.load(location)

    # onnx check ok
    onnx.checker.check_model(model)

    # onnx runtime session run ok
    sess = onnxruntime.InferenceSession(location)
    input_ids = np.array([[1346, 7541, 287, 15489, 1989, 13]], dtype=np.int64)
    attention_mask = np.array([[1., 1., 1., 1., 1., 1.]], dtype=np.float32)
    out_token_num = np.array([30], dtype=np.int64)
    result = sess.run(None, {'input_ids': input_ids,
                             'attention_mask': attention_mask,
                             'out_token_num': out_token_num})
    print(result)

    # symbolic shape inference fails
    out_mp = sym.SymbolicShapeInference.infer_shapes(in_mp=model, auto_merge=True)
```

- Attach the ONNX model to the issue (where applicable) to expedite investigation.

you can check model at https://github.com/tzhang-666/issues/blob/master/gpt2_full_noPast_wtoken.zip

**Expected behavior**
The ONNX Runtime should be able to propagate the shape and dimension information across the entire model.

**Screenshots**
```
Traceback (most recent call last):
  File "D:/gpt2infer.py", line 25, in <module>
    out_mp = sym.SymbolicShapeInference.infer_shapes(in_mp=model, auto_merge=True)
  File "C:\Users\u\Miniconda3\envs\cpu_env\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 1553, in infer_shapes
    all_shapes_inferred = symbolic_shape_inference._infer_impl()
  File "C:\Users\u\Miniconda3\envs\cpu_env\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 1403, in _infer_impl
    self.dispatcher_[node.op_type](node)
  File "C:\Users\u\Miniconda3\envs\cpu_env\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 804, in _infer_Loop
    self._onnx_infer_subgraph(node, subgraph)
  File "C:\Users\u\Miniconda3\envs\cpu_env\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 395, in _onnx_infer_subgraph
    all_shapes_inferred = symbolic_shape_inference._infer_impl(self.sympy_data_.copy())
  File "C:\Users\u\Miniconda3\envs\cpu_env\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 1403, in _infer_impl
    self.dispatcher_[node.op_type](node)
  File "C:\Users\u\Miniconda3\envs\cpu_env\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 1239, in _infer_Tile
    repeats_value = self._get_value(node, 1)
  File "C:\Users\u\Miniconda3\envs\cpu_env\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 310, in _get_value
    assert name in self.sympy_data_ or name in self.initializers_
AssertionError
```

**Additional context**
N/A

**Comment by KeDengMS** [comment]
Thanks for reporting this. Seems like a bug that's fixed in #8280. Could you give it a try?

**Comment by tzhang-666** [comment]
Thanks KeDengMS, your method works!
